### PR TITLE
[AIRFLOW-1359] Use default_args in CloudML eval

### PR DESCRIPTION
**Pull request by peterjdolan**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-XXX

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:

This change makes the create_evaluate_ops utility method make use of the
default_args parameters of the DAG when possible. This simplifies the
usage of the create_evaluate_ops method, and improves the usefulness of
a variety of default_args.

To further the usefulness of default_args for Cloud ML Operators, this
change also introduces version_name to the CloudMLVersionOperator,
allowing model_name and version_name to be specified across an entire
pipeline.

This change also resolves a small TODO by making the
DataFlowPythonOperator's `options` and `dataflow_default_options`
variables templatized.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by bolkedebruin** [comment]
Can you update your commit message slightly? From the title I wasn't able to deduct your were changing CloudML* and I had the idea your were changing core functionality.

**Comment by peterjdolan** [comment]
Updated commit and PR title.

**Comment by peterjdolan** [comment]
Rebased onto master to resolve unrelated failing tests.

**Comment by criccomini** [comment]
@N3da @jiwang576 @leomzhong can one or more of you guys +1 this?

**Comment by criccomini** [comment]
LGTM, but awaiting a +1 from one of the other owners on this file

**Comment by peterjdolan** [comment]
@yk5 would be the most appropriate person to review.

**Comment by criccomini** [comment]
Sounds good! I'll let it simmer for a day. If no response.... MERGE :D

File: airflow/contrib/operators/cloudml_operator_utils.py [hunk 1]
```diff
@@ -149,9 +128,46 @@ def validate_err_and_count(summary):
...
+    :param model_name: Used to indicate a model to use for prediction. Can be
+        used in combination with version_name, but cannot be used together with
+        model_uri. See CloudMLBatchPredictionOperator for more detail. If None,
+        then the `dag`'s `default_arguments['model_name']` will be used.
+    :type model_name: string
+
+    :param version_name: Used to indicate a model version to use for prediciton,
+        in combination with model_name. Cannot be used together with model_uri.
+        See CloudMLBatchPredictionOperator for more detail. If None, then the
+        `dag`'s `default_arguments['version_name']` will be used.
```
Comment (yk5) [inline airflow/contrib/operators/cloudml_operator_utils.py]: DocString keeps referring to 'default_arguments' but the code is using 'default_args'. Fix either.

File: airflow/contrib/operators/cloudml_operator_utils.py [hunk 2]
```diff
@@ -170,10 +186,20 @@ def validate_err_and_count(summary):
     if not callable(validate_fn):
         raise AirflowException("`validate_fn` param must be callable.")
 
+    if dag is not None:
+        default_args = dag.default_args
+        if default_args is not None:
```
Comment (yk5) [inline airflow/contrib/operators/cloudml_operator_utils.py]: Maybe you could merge two ifs into one,
if dag is not None and dag.default_args is not None:
  default_args = dag.default_args
  ...

**Review by yk5 — COMMENTED** [review]
LG.

File: airflow/contrib/operators/cloudml_operator_utils.py (hunk 1 above)
Comment (peterjdolan) [reply airflow/contrib/operators/cloudml_operator_utils.py]: Done.

File: airflow/contrib/operators/cloudml_operator_utils.py (hunk 2 above)
Comment (peterjdolan) [reply airflow/contrib/operators/cloudml_operator_utils.py]: Done.

**Pull request closed without merging**